pkgs/servers/search/manticoresearch/default.nix

```diff
@@ -0,0 +1,70 @@
+{ lib, stdenv, fetchFromGitHub, fetchurl
+, bison, cmake, flex, pkg-config
+, boost, icu, libstemmer, mariadb-connector-c, re2
+}:
+let
+  columnar = stdenv.mkDerivation rec {
+    pname = "columnar";
+    version = "c16-s5"; # see NEED_COLUMNAR_API/NEED_SECONDARY_API in Manticore's GetColumnar.cmake
+    src = fetchFromGitHub {
+      owner = "manticoresoftware";
+      repo = "columnar";
+      rev = version;
+      sha256 = "sha256-iHB82FeA0rq9eRuDzY+AT/MiaRIGETsnkNPCqKRXgq8=";
+    };
+    nativeBuildInputs = [ cmake ];
+    cmakeFlags = [ "-DAPI_ONLY=ON" ];
+    meta = {
+      description = "A column-oriented storage and secondary indexing library";
+      homepage = "https://github.com/manticoresoftware/columnar/";
+      license = lib.licenses.asl20;
+      platforms = lib.platforms.all;
+    };
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "manticoresearch";
+  version = "5.0.3";
+
+  src = fetchFromGitHub {
+    owner = "manticoresoftware";
+    repo = "manticoresearch";
+    rev = version;
+    sha256 = "sha256-samZYwDYgI9jQ7jcoMlpxulSFwmqyt5bkxG+WZ9eXuk=";
+  };
+
+  nativeBuildInputs = [
+    bison
+    cmake
+    flex
+    pkg-config
+  ];
+
+  buildInputs = [
+    boost
+    columnar
+    icu.dev
+    libstemmer
+    mariadb-connector-c
+    re2
+  ];
+
+  postPatch = ''
+    sed -i 's/set ( Boost_USE_STATIC_LIBS ON )/set ( Boost_USE_STATIC_LIBS OFF )/' src/CMakeLists.txt
```

Comment:
If it possible I would still configure this as cmakeFlag since this is a option we can use and don't have to patch

Reply:
I’m not good at cmake, but AIUI `SET(Foo ON)` will see Foo even if it’s already been specified from a flag. Doesn’t it need CACHE in order to be override from the cmake flags? Or am I missing something?